staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -614,15 +614,14 @@ func (r *Request) Watch(ctx context.Context) (watch.Interface, error) {
 	}
 	url := r.URL().String()
 	for {
-		req, err := r.newHTTPRequest(ctx)
-		if err != nil {
-			return nil, err
-		}
-
 		if err := r.retry.Before(ctx, r); err != nil {
 			return nil, r.retry.WrapPreviousError(err)
 		}
 
+		req, err := r.newHTTPRequest(ctx)
```

Comment:
The addition of `.retry` to the Request object broke threadsafety, and so if we have to remove `retry` this refactor doesn't make sense.  Will come back after discussing this.

Reply:
I think thread safety is probably not a release blocker (previous regression), unlike the body close race condition.  Therefore we don't have to block this fix on a refactor of `retry`, the refactor will simply use this logic.